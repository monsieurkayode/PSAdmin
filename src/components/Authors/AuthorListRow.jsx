import React from 'react';
import { Link } from 'react-router-dom';
import { string, func } from 'prop-types';

const AuthorListRow = ({
  id,
  firstName,
  lastName,
  deleteAuthor
}) => (
  <tr>
    <td>&nbsp;</td>
    <td><Link to={`/author/${id}`}>{id}</Link></td>
    <td>{firstName}&nbsp;{lastName}</td>
    <td>
      <button
        type="button"
        className="btn btn-sm btn-danger"
        onClick={() => deleteAuthor(id)}
      >
        Delete
      </button>
    </td>
  </tr>
);

AuthorListRow.propTypes = {
  id: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired,
  deleteAuthor: func.isRequired
};

export default AuthorListRow;
