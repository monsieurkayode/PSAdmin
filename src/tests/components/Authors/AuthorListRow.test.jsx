import React from 'react';
import { shallow } from 'enzyme';
import AuthorListRow from '../../../components/Authors/AuthorListRow';
import mockAuthor from '../../__mocks__/mockAuthor';

const setup = () => {
  const props = {
    ...mockAuthor,
    deleteAuthor: jest.fn()
  };

  const wrapper = shallow(<AuthorListRow {...props} />);

  return {
    props,
    wrapper
  };
};

describe('AuthorListRow Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a table row with four table data cells', () => {
    expect(wrapper.find('tr').length).toBe(1);
    expect(wrapper.find('td').length).toBe(4);
  });

  it('renders a Link component', () => {
    expect(wrapper.find('Link').length).toBe(1);
  });

  it('renders a Delete button', () => {
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button').text()).toBe('Delete');
  });

  it('calls deleteCourse onClick of Delete button', () => {
    const deleteButton = wrapper.find('button');
    deleteButton.simulate('click');

    expect(props.deleteAuthor).toHaveBeenCalled();
    expect(props.deleteAuthor).toHaveBeenLastCalledWith(mockAuthor.id);
  });
});
