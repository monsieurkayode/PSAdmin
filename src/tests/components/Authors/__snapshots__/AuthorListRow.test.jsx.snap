// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorListRow Component renders correctly 1`] = `
<tr>
  <td>
     
  </td>
  <td>
    <Link
      replace={false}
      to="/author/cory-house"
    >
      cory-house
    </Link>
  </td>
  <td>
    Cory
     
    House
  </td>
  <td>
    <button
      className="btn btn-sm btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
